src/PooledArrays.jl

```diff
@@ -137,6 +139,10 @@ PooledArray(t::Type, r::Type) = PooledArray(Array(t,0), r)
 ##
 ##############################################################################
 
+DataAPI.refarray(pa::PooledArray) = pa.refs
+DataAPI.refvalue(pa::PooledArray, i) = pa.pool[i]
```

Comment:
Maybe restrict this to `Integer`?

Reply:
Good point, it should be an integer as it's used as an index to a vector.